paper-api/src/main/java/org/bukkit/MusicInstrument.java

```diff
@@ -48,25 +50,50 @@ public static MusicInstrument getByKey(@NotNull NamespacedKey namespacedKey) {
      * @return the memoryKeys
      * @deprecated use {@link Registry#iterator()}.
      */
-    @NotNull
     @Deprecated(since = "1.20.1")
     public static Collection<MusicInstrument> values() {
         return Collections.unmodifiableCollection(Lists.newArrayList(Registry.INSTRUMENT));
     }
 
-    @NotNull
-    private static MusicInstrument getInstrument(@NotNull String key) {
+    private static MusicInstrument getInstrument(final String key) {
         return RegistryAccess.registryAccess().getRegistry(RegistryKey.INSTRUMENT).getOrThrow(NamespacedKey.minecraft(key));
     }
 
-    // Paper start - deprecate getKey
+    /**
+     * Gets for how long the use duration is for the instrument.
+     *
+     * @return the duration.
+     */
+    public abstract float getDuration();
+
+    /**
+     * Gets the range of the sound.
+     *
+     * @return the range of the sound.
+     */
+    public abstract float getRange();
+
+    /**
+     * Provides the description of this instrument as displayed to the client.
+     *
+     * @return the description component.
+     */
+    public abstract Component description();
+
+    /**
+     * Gets the sound/sound-event for this instrument.
+     *
+     * @return a sound
+     */
+    public abstract Sound getSoundEvent();
```

Comment:
I'd just call it getSound instead of SoundEvent

Reply:
Hmm just the method not?
I set this name for avoid confusing for what this get and make consist with the create instrument PR.